src/client/datascience/jupyter/kernels/kernelSelections.ts

```diff
@@ -95,27 +95,35 @@ export class KernelSelectionProvider {
      */
     public async getKernelSelectionsForRemoteSession(
         resource: Resource,
-        sessionManager: IJupyterSessionManager,
+        sessionManagerCreator: () => Promise<IJupyterSessionManager>,
```

Comment:
We cannot give it a session manager, we need to pass a factory so that this function can manage the lifetime of the session manager object.

Reply:
Previously we'd return the cached item while in the background we'd retrieve the kernels from session manager, but the calling code would see that the promise was resolved hence the session manager would be disposed.
Hence calling code cannot own the lifetime of the session manager.